ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/EJBClientDescriptorMetaDataProcessor.java

```diff
@@ -133,6 +136,19 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
         if (ejbClientDescriptorMetaData == null && ! interceptorsDefined) {
             return;
         }
+
+        final CapabilityServiceSupport support = deploymentUnit.getAttachment(org.jboss.as.server.deployment.Attachments.CAPABILITY_SERVICE_SUPPORT);
+        if (ejbClientDescriptorMetaData != null && ejbClientDescriptorMetaData.getProfile() == null) {
+                for (EJBClientDescriptorMetaData.RemotingReceiverConfiguration receiverConfiguration : ejbClientDescriptorMetaData.getRemotingReceiverConfigurations()) {
+                    final String requiredCap = RuntimeCapability.buildDynamicCapabilityName(OUTBOUND_CONNECTION_CAPABILITY_NAME, receiverConfiguration.getOutboundConnectionRef());
+                    if (!support.hasCapability(requiredCap)) {
+                        throw EjbLogger.DEPLOYMENT_LOGGER.deploymentRequiresCapability(
+                                deploymentUnit.getName(),
+                                requiredCap);
+                    }
+            }
+        }
+
```

Comment:
Is this validation necessary?  This isn't a management model consistency check, where we need to fail in Stage.MODEL if something is wrong. This is runtime stuff. If the reference is invalid won't MSC report the service wiring failure? I'm reluctant to start adding additional validation. It has the potential to produce nicer failures (although a "missing capability" message isn't particularly nicer than a "missing service" message) but it's either going to end up being done in a few random spots or it will be a very large addition of potentially fragile code.

Reply:
It is unnecessary, the intention was fail as soon as possible with a nicer message.
I agree on your points so I'll remove this and let the missing service message in case a user runs into this situation.